internal/policy/verify_test.go

```diff
@@ -744,39 +744,65 @@ func TestVerifyCommit(t *testing.T) {
+		assert.Equal(t, expectedStatus, status)
 
-	entry = rsl.NewReferenceEntry(string(plumbing.NewTagReferenceName(tagName)), tagID)
-	entryID = common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
-	entry.ID = entryID
+		entry = rsl.NewReferenceEntry(string(plumbing.NewTagReferenceName(tagName)), tagID)
+		entryID = common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
+		entry.ID = entryID
 
-	// Use tag ID
-	expectedStatus = map[string]string{tagID.String(): goodTagSignatureMessage}
-	status = VerifyTag(context.Background(), repo, []string{tagID.String()})
-	assert.Equal(t, expectedStatus, status)
+		// Use tag ID
+		expectedStatus = map[string]string{tagID.String(): goodTagSignatureMessage}
+		status = VerifyTag(context.Background(), repo, []string{tagID.String()})
+		assert.Equal(t, expectedStatus, status)
 
-	// Use tagName
-	expectedStatus = map[string]string{tagName: goodTagSignatureMessage}
-	status = VerifyTag(context.Background(), repo, []string{tagName})
-	assert.Equal(t, expectedStatus, status)
+		// Use tagName
+		expectedStatus = map[string]string{tagName: goodTagSignatureMessage}
+		status = VerifyTag(context.Background(), repo, []string{tagName})
+		assert.Equal(t, expectedStatus, status)
 
-	// Use refs path for tagName
-	expectedStatus = map[string]string{string(plumbing.NewTagReferenceName(tagName)): goodTagSignatureMessage}
-	status = VerifyTag(context.Background(), repo, []string{string(plumbing.NewTagReferenceName(tagName))})
-	assert.Equal(t, expectedStatus, status)
+		// Use refs path for tagName
+		expectedStatus = map[string]string{string(plumbing.NewTagReferenceName(tagName)): goodTagSignatureMessage}
+		status = VerifyTag(context.Background(), repo, []string{string(plumbing.NewTagReferenceName(tagName))})
+		assert.Equal(t, expectedStatus, status)
+	})
+
+	t.Run("tag verification with changed tag", func(t *testing.T) {
+		repo, _ := createTestRepository(t, createTestStateWithPolicy)
+		refName := "refs/heads/main"
+
+		commitIDs := common.AddNTestCommitsToSpecifiedRef(t, repo, refName, 3, gpgKeyBytes)
+		entry := rsl.NewReferenceEntry(refName, commitIDs[len(commitIDs)-1])
+		entryID := common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
+		entry.ID = entryID
+
+		tagName := "v1"
+		tagID := common.CreateTestSignedTag(t, repo, tagName, commitIDs[len(commitIDs)-1], gpgKeyBytes)
+
+		entry = rsl.NewReferenceEntry(string(plumbing.NewTagReferenceName(tagName)), tagID)
+		entryID = common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
+		entry.ID = entryID
+
+		common.CreateTestSignedTag(t, repo, tagName, commitIDs[len(commitIDs)-2], gpgKeyBytes)
+
+		// Use tag ID
+		expectedStatus := map[string]string{tagID.String(): fmt.Sprintf("verifying RSL entry failed, %s", ErrUnauthorizedSignature.Error())}
+		status := VerifyTag(context.Background(), repo, []string{tagID.String()})
+		assert.Equal(t, expectedStatus, status)
+	})
```

Comment:
This doesn't include a test for duplicate checks in the verification flow, right?

Reply:
Sorry, could you clarify what you mean?

Reply:
recording tag v1 in the RSL and then recording the exact same tag again in the RSL. The RSL should enforce immutability of tags it's aware of.

Reply:
I have added this test